src/logic/function/basic.lean

```diff
@@ -92,6 +92,10 @@ lemma injective.of_comp_iff' (f : α → β) {g : γ → α} (hg : bijective g)
     hx ▸ hy ▸ λ hf, h hf ▸ rfl,
   λ h, h.comp hg.injective⟩
 
+lemma injective.of_subsingleton {α β : Type*} [h : subsingleton α] (f : α → β) :
+  function.injective f :=
+λ a b ab, (eq_iff_true_of_subsingleton _ _).mpr trivial
```

Comment:
golfed:
```suggestion
lemma injective_of_subsingleton {α β : Sort*} [subsingleton α] (f : α → β) :
  function.injective f :=
λ a b ab, subsingleton.elim _ _
```
No dot notation either, since there is no `injective` argument to make it useful.